### PR TITLE
[MODORDERS-970] - Do not overwrite Item fields when piece fields are null/empty

### PR DESCRIPTION
**Pull request by imerabishvili**
## Purpose
https://issues.folio.org/browse/MODORDERS-970

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Examples exist for all schemas
  - [ ] Descriptions exist for all schema properties
  - [ ] All schemas pass raml-cop linting
- Does this introduce breaking changes?
  - [ ] Were there any schema changes?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: mod-orders-storage/schemas/piece.json [hunk 1]
```diff
@@ -54,6 +54,18 @@
...
+      "description": "Barcode is the descriptive information for the barcode of a serial. Synchronized with inventory item."
+    },
+    "accessionNumber": {
+      "type": "string",
+      "description": "AccessionNumber is the descriptive information for the accession number of a serial. Synchronized with inventory item."
+    },
+    "itemLevelCallNumber": {
+      "type": "string",
+      "description": "ItemLevelCallNumber is the descriptive information for the call number of a serial. Synchronized with inventory item."
+    },
```
Comment (SerhiiNosko) [inline mod-orders-storage/schemas/piece.json:68]: please update samples as well

Comment (imerabishvili) [reply mod-orders-storage/schemas/piece.json:68]: done

File: mod-orders-storage/examples/piece_collection.sample [hunk 2]
```diff
@@ -13,6 +13,9 @@
         "displayOnHolding": false,
         "enumeration": "Volume 1",
         "chronology": "2021",
+      "barcode": "0987654111",
+      "accessionNumber": "2015.1",
+      "itemLevelCallNumber": "c123",
```
Comment (SerhiiNosko) [inline mod-orders-storage/examples/piece_collection.sample:18]: formatting should be fixed

Comment (imerabishvili) [reply mod-orders-storage/examples/piece_collection.sample]: fixed formatting and naming

File: mod-orders-storage/schemas/piece.json [hunk 3]
```diff
@@ -54,6 +54,18 @@
...
     },
+    "barcode": {
+      "type": "string",
+      "description": "Barcode is the descriptive information for the barcode of a serial. Synchronized with inventory item."
+    },
+    "accessionNumber": {
+      "type": "string",
+      "description": "AccessionNumber is the descriptive information for the accession number of a serial. Synchronized with inventory item."
+    },
+    "callNumber": {
```
Comment (SerhiiNosko) [inline mod-orders-storage/schemas/piece.json:65]: In Item schema this field is named as: itemLevelCallNumber, maybe @usavkov-epam knows the correct name

**Review by SerhiiNosko — APPROVED** [review]

**Review by usavkov-epam — APPROVED** [review]

**Pull request merged**